extensions/amazon-lambda-rest/runtime/src/main/java/io/quarkus/amazon/lambda/http/graal/LambdaContainerHandlerSubstitution.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.amazon.lambda.http.graal;
+
+import com.amazonaws.serverless.proxy.internal.LambdaContainerHandler;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(LambdaContainerHandler.class)
+public class LambdaContainerHandlerSubstitution {
+
+    // afterburner does not work in native mode, so let's ensure it's never registered
+    @Substitute
+    private static void registerAfterBurner() {
+
```

Comment:
Should we log something? I have no idea if people would expect it to be around?

Reply:
I don't think they expect it to, unless they look at the code of `LambdaContainerHandler `. And in any case, afterburner doesn't change the functionality in any way, it just tries to make serialization and deserialization faster

Reply:
TBH, if Afterburner doesn't work in native, I'm not sure we should use it in JVM mode either. Could hide some issues.

Not my ballpark though so I suppose I will let @patriot1burke decide.

Reply:
I think we should merge this for now and we can certainly improve if needed

Reply:
We've been waiting for this a long time

Reply:
Wasn't a new aws library required to allow this fix?  I don't see an aws library bump in this PR, did it happen earlier?

Reply:
We bumped to version 1.6 (from 1.3.1 I think) which contains the change we needed.
See https://github.com/quarkusio/quarkus/pull/19040